Velaptor/OpenGL/Buffers/RectGPUBuffer.cs

```diff
@@ -13,32 +13,47 @@ namespace Velaptor.OpenGL.Buffers;
 using Velaptor.NativeInterop.OpenGL;
 using Exceptions;
 using GPUData;
+using Guards;
 using Reactables.Core;
 using Reactables.ReactableData;
 
 /// <summary>
 /// Updates data in the rectangle GPU buffer.
 /// </summary>
 [GPUBufferName("Rectangle")]
-[BatchSize(IRenderer.BatchSize)]
 internal sealed class RectGPUBuffer : GPUBufferBase<RectBatchItem>
 {
     private const string BufferNotInitMsg = "The rectangle buffer has not been initialized.";
+    private readonly IDisposable unsubscriber;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="RectGPUBuffer"/> class.
     /// </summary>
     /// <param name="gl">Invokes OpenGL functions.</param>
     /// <param name="openGLService">Provides OpenGL related helper methods.</param>
     /// <param name="glInitReactable">Receives a notification when OpenGL has been initialized.</param>
+    /// <param name="batchSizeReactable">Receives a push notification about the batch size.</param>
     /// <param name="shutDownReactable">Receives a notification that the application is shutting down.</param>
+    /// <exception cref="ArgumentNullException">
+    ///     Invoked when any of the parameters are null.
+    /// </exception>
     public RectGPUBuffer(
         IGLInvoker gl,
         IOpenGLService openGLService,
         IReactable<GLInitData> glInitReactable,
+        IReactable<BatchSizeData> batchSizeReactable,
         IReactable<ShutDownData> shutDownReactable)
         : base(gl, openGLService, glInitReactable, shutDownReactable)
     {
+        // TODO: Test for null batch reactable and subscription
```

Comment:
Do you still need this TODO?
```suggestion
        // TODO: Test for null batch reactable and subscription
```